File: app/src/main/java/com/sliebald/cula/data/database/DateConverter.java
package com.sliebald.cula.data.database;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

@SuppressWarnings("WeakerAccess")
public class DateConverter {
    @TypeConverter
    public static Date toDate(Long timestamp) {
        return timestamp == null ? null : new Date(timestamp);
    }

    @TypeConverter
    public static Long toTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }
}

File: app/src/main/java/com/sliebald/cula/data/database/DateConverter.kt
package com.sliebald.cula.data.database

import androidx.room.TypeConverter
import java.util.*

/**
 * Custom converter for date/time.
 */
class DateConverter {

    @TypeConverter
    fun timestampToDate(timestamp: Long): Date {
        return Date(timestamp)
    }

    @TypeConverter
    fun dateToTimestamp(date: Date): Long {
        return date.time
    }
}